lib/bom.js

```diff
@@ -67,57 +66,116 @@ type Gamepad = {
   timestamp: number;
 }
 
+// deprecated
 type BatteryManager = {
-    charging: boolean;
-    chargingTime: number;
-    dischargingTime: number;
-    level: number;
-    onchargingchange: ?Function;
-    onchargingtimechange: ?Function;
-    ondischargingtimechange: ?Function;
-    onlevelchange: ?Function;
+  +charging: boolean;
+  +chargingTime: number;
+  +dischargingTime: number;
+  +level: number;
+  onchargingchange: ?((event: any) => mixed);
+  onchargingtimechange: ?((event: any) => mixed);
+  ondischargingtimechange: ?((event: any) => mixed);
+  onlevelchange: ?((event: any) => mixed);
 }
 
 // https://wicg.github.io/web-share
 type ShareData = {
-    title?: string;
-    text?: string;
-    url?: string;
+  title?: string;
+  text?: string;
+  url?: string;
 }
 
-declare class NavigatorCommon {
+type PermissionName = "geolocation" | "notifications" | "push" | "midi" | "camera" | "microphone" | "speaker" | "device-info" | "background-sync" | "bluetooth" | "persistent-storage" | "ambient-light-sensor" | "accelerometer" | "gyroscope" | "magnetometer" | "clipboard";
```

Comment:
"clipboard" should be "clipboard-read" | "clipboard-write" according to https://w3c.github.io/clipboard-apis/#clipboard-permissions and https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API. and its own descriptor, I suppose

Reply:
Maybe I'm missing something, but in https://w3c.github.io/permissions/#enumdef-permissionname there isn't "clipboard-read" and "clipboard-write"

Reply:
Yeah, I tried using this API, and there isn't "clipboard"